### ISSUE TITLE
Idea to develop VS Code plugin with for linting tl;dr pages automatically

### ISSUE THREAD

**Issue by EmilySeville7cfg** [tldr-pages/tldr#7878]
## Description

It would be nice to have such thing because we could see our errors instantly and fix them (probably automatically).

## Linting process

Linting could be done while we are typing our tl;dr pages not just on file save. But I think we have to allow user to choose when to lint pages via settings in `.vscode/settings.json`.

**Comment by blueskyson** [comment]
Good idea.

**Comment by SethFalco** [comment]
How do you intend to match tldr pages though, or will you only support this repository? :thinking: 

For example, people can make their own pages, or may wish to just write the page in their editor and paste it into GitHub in the web editor. (So they have tldr pages, but now our repos structure.)

Since our files just have `.md`, not something like `.tldr.md`, there is little to separate them from any other Markdown files.

Could you define here what your expectations for such an extension are in the issue?

**Comment by EmilySeville7cfg** [comment]
> How do you intend to match tldr pages though, or will you only support this repository? thinking

I think by default all markdown files which are TlDr in our repo can be considered as TlDr pages (extension should check whether it is running in git repo and whether the current document is actually is TlDr page), but it should be customizable via some editor settings. It should be possible to make extension interpret any markdown file as a TlDr page if user knows that it is.

**Comment by SethFalco** [comment]
So for example, in the case of VSC, you'd activate the extension either when `onLangauge:markdown`.

If it's this repo, then hardcore that all `*.md` in `pages*` should be annotated by tldr-lint.
If it's not this repo, there should be a command in the command pallete to lint the currently open file?

Is that right?

---

If so, this sounds a bit mostly alright but might be a bit much. I wonder is there a way to make do this lighter where it's just a configured in Prettier or ESLint, basically using established tools and extensions instead of making our own. :thinking: 

If you really want to take this approach though, then I think the best way would just be making a command that the user can execute manually and that's it. Someone who installs the plugin would know they have to do the command from the README, and it avoids making an extension which will activate and process files redundantly 99% of the time.

This also makes it to use with any arbitrary text or file format, not just `.md`.

Just some thoughts though, you can do what you want. I'm not saying your proposal is bad, I just personally like to minimize extension activation, and by the sounds of it, this will activate for every single workspace and process every Markdown file I open, when I know I only need it for the tldr repo.

**Comment by EmilySeville7cfg** [comment]
> So for example, in the case of VSC, you'd activate the extension with onLangauge:markdown?

Yes.

> If it's this repo, then hardcore that all *.md in pages* should be annotated by tldr-lint.
If it's not this repo, there should be a command in the command pallete to lint the currently open file? Or it's just going to process the file to determine if it's a page (though note this may be prone to false positives) and add the annotations from there?
>
> Is that right?

Right. :)

**Comment by kbdharun** [comment]
I agree with @SethFalco. Automatic triggering on all markdown files would be troublesome and will conflict with markdown lint. How about triggering with a keyboard shortcut or key binding similar to other extensions? 

Just be sure not to use these keybindings https://code.visualstudio.com/shortcuts/keyboard-shortcuts-windows.pdf.
 
I would suggest something like `Ctrl` + `T` + `R` as `CTRL` + `T` is already used for showing symbols.

**Comment by EmilySeville7cfg** [comment]
> How about triggering with a keyboard shortcut or key binding similar to other extensions?

Nice idea. 😄 (if this behavior can be customized)

**Comment by SethFalco** [comment]
> For me it's better to lint pages automatically as TlDr project

I agree doing it automatically is better, you can always find ways to improve one aspect of the UX if you're willing to degrade other aspects. I think the con of falsely doing it 99% of the time when it won't achieve anything is worse.

If you could come up with a way to do it automatically, with 100% certain you're doing rightfully so, then cool.

But otherwise I think registering a command and shortcut is better. But once again, I'm just saying my own preferences.

**Comment by EmilySeville7cfg** [comment]
@SethFalco, can you [check](https://github.com/tldr-pages/tldr-emacs-extension/tree/feature/refresh-available-actions-and-add-new) this extension (if you are Emacs user)?

**Comment by SethFalco** [comment]
I only use VSC and Vim. ^-^'

**Comment by kbdharun** [comment]
- I only use **VSC** and Vim. ^-^'

Then can you check this out https://github.com/tldr-pages/tldr-code-extension?

**Comment by EmilySeville7cfg** [comment]
> I only use VSC and Vim. ^-^'

Do you wanna to develop TlDr linting plugin for Vim/Neovim? 😄

**Comment by EmilySeville7cfg** [comment]
I don't support TlDr project plugins anymore. Do with them what you want.
